test/test_dtype.py

```diff
@@ -36,12 +36,12 @@ def _assert_eq(tensor:Tensor, target_dtype:DType, target):
   if DEBUG >= 2: print(tensor.numpy())
   try:
     assert tensor.dtype == target_dtype
-    np.testing.assert_allclose(tensor.numpy(), target)
+    np.testing.assert_allclose(tensor.numpy(), target, rtol=1e-3 if target_dtype == dtypes.float16 else 1e-7)
```

Comment:
Why did these things change?

Reply:
The underlying issue that we have to use `.rz` as the rounding modifier (round towards zero), since rounding towards nearest is not supported by PTX 2.2, and thus not support by ocelot. This error is negligible in float32 but becomes apparent in float16.

Reply:
And the other change in this file is due to a weird edge case in numpy: `a.astype(np.float16).tolist()` gives values in float32 not float16.